### PR TITLE
Subscribers: Number format subscribers count for subscribers panel in editor

### PR DESCRIPTION
**Pull request by ebinnion**
#### Changes proposed in this Pull Request:

* The number of subscribers that will see a post, which is shown in the subscribers panel when publishing, currently it not formatted. This PR adds formatting.

#### Jetpack product discussion

p1653675252620949-slack-C0299DMPG

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

Screen Shot 2022-05-27 at 2 28 28 PM

Screen Shot 2022-05-27 at 2 28 36 PM

* Checkout PR
* Publish a post on a site that has more than 1,000 subscribers or change the const to a let and manually modify the number

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ import { useSelect } from '@wordpress/data';
 import { store as editorStore } from '@wordpress/editor';
 import { __, _n, sprintf } from '@wordpress/i18n';
 import { isComingSoon, isPrivateSite } from '@automattic/jetpack-shared-extension-utils';
+import { numberFormat } from '@automattic/jetpack-components';
```
Comment (mtias) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js:10]: I think we should look at a more compact display at some point, like https://github.com/Automattic/wp-calypso/tree/9fb7ac149e55361c0c3a9639ad0caa5e15fc88b0/client/lib/format-number-compact

Comment (jeherve) [reply projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js:10]: Yeah, that would be nice to implement something like this in our `numberFormat`. I've created an issue to keep track of this:
https://github.com/Automattic/jetpack/issues/24554

We purposefully got rid of the `i18n-calypso` dependency in Jetpack a while ago, so we would be implementing this a bit differently.

**Review by jeherve — APPROVED** [review]
Tests well for me. 🚢 

image

**Comment by jeherve** [comment]
r246582-wpcom

**Pull request merged**